geode-redis/src/main/java/org/apache/geode/redis/internal/executor/set/GeodeRedisSetSynchronized.java

```diff
@@ -59,8 +60,15 @@ public long srem(Collection<ByteArrayWrapper> membersToRemove) {
       Set<ByteArrayWrapper> newValue = createSet(oldValue);
       newValue.removeAll(membersToRemove);
       removedCount.set(oldValue.size() - newValue.size());
+
       return newValue;
     });
+
+    if (members().size() == 0) {
```

Comment:
Same comments on this block as the above hdel

Reply:
I will update and add a TODO.